src/main/java/hudson/plugins/sshslaves/verifiers/ManuallyProvidedKeyVerificationStrategy.java

```diff
@@ -92,8 +94,16 @@ private static HostKey parseKey(String key) {
                 RSASHA1Verify.decodeSSHRSAPublicKey(keyValue);
             } else if ("ssh-dss".equals(algorithm)) {
                 DSASHA1Verify.decodeSSHDSAPublicKey(keyValue);
+            } else if ("ssh-ed25519".equals(algorithm)) {
```

Comment:
You have to invoke the methods via reflection as well. This should work:
```java
    private static HostKey parseKey(String key) {
        if (!key.contains(" ")) {
            throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_TwoPartKey());
        }
        StringTokenizer tokenizer = new StringTokenizer(key, " ");
        String algorithm = tokenizer.nextToken();
        byte[] keyValue = Base64.decode(tokenizer.nextToken());
        if (null == keyValue) {
            throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_Base64EncodedKeyValueRequired());
        }
        
        try {
            if ("ssh-rsa".equals(algorithm)) {
                RSASHA1Verify.decodeSSHRSAPublicKey(keyValue);
            } else if ("ssh-dss".equals(algorithm)) {
                DSASHA1Verify.decodeSSHDSAPublicKey(keyValue);
            } else if (isClass("com.trilead.ssh2.signature.ED25519KeyAlgorithm") && "ssh-ed25519".equals(algorithm)) {
                decodePublicKey("com.trilead.ssh2.signature.ED25519KeyAlgorithm", keyValue);
            } else if (isClass("com.trilead.ssh2.signature.ECDSAKeyAlgorithm") && "ecdsa-sha2-nistp256".equals(algorithm)) {
                decodePublicKey("com.trilead.ssh2.signature.ECDSAKeyAlgorithm", keyValue);
            } else if (isClass("com.trilead.ssh2.signature.ECDSAKeyAlgorithm") && "ecdsa-sha2-nistp384".equals(algorithm)) {
                decodePublicKey("com.trilead.ssh2.signature.ECDSAKeyAlgorithm", keyValue);
            } else if (isClass("ECDSAKeyAlgorithm") && "ecdsa-sha2-nistp521".equals(algorithm)) {
                decodePublicKey("com.trilead.ssh2.signature.ECDSAKeyAlgorithm", keyValue);
            } else {
                throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_UnknownKeyAlgorithm());
            }
        } catch (IOException ex) {
            throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_KeyValueDoesNotParse(algorithm), ex);
        }  catch (StringIndexOutOfBoundsException ex) {
            // can happen in DSASHA1Verifier with certain values (from quick testing)
            throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_KeyValueDoesNotParse(algorithm), ex);
        } catch (NoSuchMethodException e) {
            throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_KeyValueDoesNotParse(algorithm), ex);
        } catch (IllegalAccessException e) {
            throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_KeyValueDoesNotParse(algorithm), ex);
        } catch (InvocationTargetException e) {
            throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_KeyValueDoesNotParse(algorithm), ex);
        } catch (ClassNotFoundException e) {
            throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_KeyValueDoesNotParse(algorithm), ex);
        }

        return new HostKey(algorithm, keyValue);
    }

    private static void decodePublicKey(String className, byte[] keyValue) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, IllegalAccessException {
        Class.forName(className).getMethod("decodePublicKey", byte[].class).invoke(keyValue);
    }
```

Please note also the changes of the `isClass` calls. Regarding the Javadoc of `Class.forName()` you have the pass the fully qualified name of the class. Btw. I think it's time to switch the java level from 6 to 7 (pom.xml). So multi-catches will be possible.